utf8parse/src/lib.rs

```diff
@@ -91,3 +91,41 @@ impl Parser {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io;
+    use std::io::prelude::*;
+    use std::fs::File;
+    use Receiver;
+    use Parser;
+
+    impl Receiver for String {
+        fn codepoint(&mut self, c: char) {
+            self.push(c);
+        }
+
+        fn invalid_sequence(&mut self) {
+        }
+    }
+
+    #[test]
+    fn utf8parse_test() {
+        let mut buffer = String::new();
+        let mut file = File::open("UTF-8-demo.txt").unwrap();
+        let mut file_to_string = file.read_to_string(&mut buffer).unwrap();
+        let parser = Parser::new();
+
+        // expected struct `std::vec::Vec`, found usize
+        let expected = String::from_utf8(file_to_string).unwrap();
+
+        // no method named `iter` found for type `usize` in the current scope
+        let actual = for byte in file_to_string.iter() {
+            // expected &mut _, found struct `std::string::String`
+            parser.advance(buffer, byte);
```

Comment:
There's not going to be a useful return value for this loop. The `actual` string you're producing _is_ the receiver that gets passed to `advance()`.